File: 2938-separate-black-and-white-balls/2938-separate-black-and-white-balls.java
class Solution {
    public long minimumSteps(String s) {
        var n = s.length();
        long swaps = 0;
        
        for(int left=n-1, right=n-1; left >= 0; left--){
            if(s.charAt(left) == '1'){
                swaps += (right - left);
                right--;
            }
        }

        return swaps;
    }
}

File: 2938-separate-black-and-white-balls/2938-separate-black-and-white-balls.py
class Solution:
    def minimumSteps(self, s: str) -> int:
        n, swaps = len(s), 0
        left = right = n-1

        while left >= 0:
            if s[left] == '1':
                swaps += (right - left)
                right -= 1
            left -= 1

        return swaps